File: zhuishushenqi/NewVersion/ZSThirdPartSDK/ZSThirdPartSDK/Classes/ZSLogin.m
//
//  ZSLogin.m
//  ZSThirdPartSDK
//
//  Created by caony on 2019/6/22.
//  Copyright © 2019 cj. All rights reserved.
//

#import "ZSLogin.h"
#import "ZSThirdPartSDK-Swift.h"

@interface ZSLogin ()

@end

@implementation ZSLogin

+ (instancetype)share {
    static ZSLogin * shareInstance = nil;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        shareInstance = [[self alloc] init];
    });
    return shareInstance;
}

- (instancetype)init
{
    self = [super init];
    if (self) {
        NSString * token = [[ZSThirdLogin shared] token];
        NSString * mobileToken = [ZSMobileLogin share].userInfo.token;
        if (token.length > 0) {
            self.token = token;
        } else if (mobileToken.length > 0) {
            self.token = mobileToken;
            self.mobileLogin = YES;
        }
    }
    return self;
}

- (BOOL)hasLogin {
    return self.token.length == 0;
}

- (void)logout {
    [[ZSThirdLoginStorage share] resetLocalUserInfo];
    [[ZSThirdLogin shared] logout];
}

- (NSInteger)lastLoginType {
    return [ZSThirdLoginStorage share].lastLoginType;
}

@end

File: zhuishushenqi/App/ZSLogin.swift
//
//  ZSLogin.swift
//  zhuishushenqi
//
//  Created by yung on 2018/10/20.
//  Copyright © 2018年 QS. All rights reserved.
//

import UIKit
import ZSAPI

class ZSLogin {
    
    var token:String = ""
    
    var lastLoginType = ZSThirdLoginStorage.share.lastLoginType
    
    var mobileLogin:Bool = false
    
    // 登录用户的关注与被关注者
    var followings:[ZSFollowings]?
    var followers:[ZSFollowings]?
    
    static let share = ZSLogin()
    private init() {
        if let token =  ZSThirdLogin.share.userInfo?.token {
            self.token = token
        } else if let token = ZSMobileLogin.share.userInfo?.token {
            self.mobileLogin = true
            self.token = token
        }
        if self.token.count > 0 {
            fetchFollowings()
            fetchFollowers()
        }
    }
    
    func hasLogin() ->Bool {
        if token != "" {
            return true
        }
        return false
    }
    
    func userInfo() ->ZSQQLoginResponse? {
        if let userInfo = ZSThirdLogin.share.userInfo {
            return userInfo
        } else if let userInfo = ZSMobileLogin.share.userInfo {
            return userInfo
        }
        return nil
    }
    
    func logout() {
        self.token = ""
        ZSThirdLoginStorage.share.resetLocalUserInfo()
        ZSThirdLogin.share.userInfo = nil
        ZSThirdLogin.share.wxTokenResp = nil
    }

    func login() {
        
    }
    
    
    func fetchFollowings() {
        let api = ZSAPI.userFollowings("\(self.userInfo()?.user?._id ?? "")")
        zs_get(api.path) { (json) in
            guard let followings = json?["followings"] as? [Any] else {
                return
            }
            if let followingModels = [ZSFollowings].deserialize(from: followings) as? [ZSFollowings] {
                self.followings = followingModels
            }
        }
    }
    
    func fetchFollowers() {
        let api = ZSAPI.userFollowers("\(self.userInfo()?.user?._id ?? "")")
        zs_get(api.path) { (json) in
            guard let followers = json?["followers"] as? [Any] else {
                return
            }
            if let followingModels = [ZSFollowings].deserialize(from: followers) as? [ZSFollowings] {
                self.followers = followingModels
            }
        }
    }
}
